Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanViewModel.kt

```diff
@@ -37,10 +47,10 @@ class SubmissionQRCodeScanViewModel @AssistedInject constructor() : CWAViewModel
     val registrationState = MutableLiveData(ApiRequestState.IDLE)
     val registrationError = SingleLiveEvent<CwaWebException>()
 
-    fun doDeviceRegistration() = launch {
+    private fun doDeviceRegistration(scanResult: QRScanResult) = launch {
         try {
             registrationState.postValue(ApiRequestState.STARTED)
-            SubmissionService.asyncRegisterDevice()
+            checkTestResult(SubmissionService.asyncRegisterDeviceViaGUID(scanResult.guid!!))
```

Comment:
I would prefer an explicit null-check here

Reply:
maybe this calls for a refactoring of `QRScanResult` as `guid` cannot be null if `isValid` is `true` as seen [here](https://github.com/corona-warn-app/cwa-app-android/blob/fix/harmonize-behaviour-scanning-expired-QR-EXPOSUREAPP-2700/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/service/submission/QRScanResult.kt#L7-L8)